packages/flutter_tools/lib/src/base/terminal.dart

```diff
@@ -81,6 +81,10 @@ abstract class Terminal {
   /// Whether the current terminal can display emoji.
   bool get supportsEmoji;
 
+  /// When we have a choice of styles (e.g. animated spinners), this selects the
+  /// style to use.
```

Comment:
Nit: I would actually hoist _emojiAnimations into a top level and then link to it from here so it is easy to find. You could also continue with naming the animations like you did with the comments, perhaps an enum/const for these?

```
PreferredStyle get preferredStyle => PreferredStyle.fireworks;
```

Reply:
My theory was that this could be used in other places too, e.g. if we wanted to have some fancy color scheme somewhere, or different emoji telling you that there was an update available, or whatever. Each of those may have a different number of available variants, so having an enum would make that tricky. (For example, even in this patch there's two sets of variants, one for Unicode spinners and one for ASCII spinners.)

The goal here was to make testing predictable, by having the test Terminal implementations always return 0.

Reply:
Yeah that makes sense to me